rct229/ruleset_functions/baseline_systems/is_baseline_system_3.py

```diff
@@ -0,0 +1,110 @@
+    HVAC_SYS,
+    find_exactly_one_hvac_system,
+)
+
+HEATING_SYSTEM = schema_enums["HeatingSystemOptions"]
+
+
+def is_baseline_system_3(rmi_b, hvac_b_id, terminal_unit_id_list, zone_id_list):
+    """
+    Get either Sys-3, Sys-3a, Sys-3b, Sys-3c, or Not_Sys_3 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 3 (PSZ), system 3a (system 3 with purchased CHW),
+    system 3b (system 3 with purchased heating), system 3c (system 3 with purchased CHW and purchased HW).
+
+    Parameters
+    ----------
+    rmi_b json
+        To evaluate if the hvac system is modeled as either Sys-11.1, Sys-11.1a, Sys-11b, Sys-11c, or Not_Sys_11.1 in the B_RMI.
+
+    hvac_b_id list
+        The id of the hvac system to evaluate.
+
+    terminal_unit_id_list
+        List list of terminal unit IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    zone_id_list list
+        list of zone IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    Returns
+    -------
+    The function returns either Sys-3, Sys-3a, Sys-3b, Sys-3c, or Not_Sys_3 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 3 (PSZ),
+    system 3a (system 3 with purchased CHW), system 3b (system 3 with purchased heating), system 3c (system 3 with purchased CHW and purchased HW).
+    """
+    is_baseline_system_3 = HVAC_SYS.UNMATCHED
+
+    hvac_b = find_exactly_one_hvac_system(rmi_b, hvac_b_id)
+    # check if the hvac system has the required sub systems for system type 3
+    has_required_sys = (
+        hvac_b.get("preheat_system") is None
+        or hvac_b["preheat_system"].get("heating_system_type") is None
+        or hvac_b["preheat_system"]["heating_system_type"] == HEATING_SYSTEM.NONE
+    )
+
+    are_sys_data_matched = (
+        # short-circuit the logic if no required data is found.
+        has_required_sys
+        # sub functions handles missing required sys, and return False.
+        and is_hvac_sys_fan_sys_cv(rmi_b, hvac_b_id)
+        and does_hvac_system_serve_single_zone(rmi_b, zone_id_list)
+        and does_each_zone_have_only_one_terminal(rmi_b, zone_id_list)
+        and are_all_terminal_heat_sources_none_or_null(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_cool_sources_none_or_null(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_fans_null(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_types_cav(rmi_b, terminal_unit_id_list)
+    )
+
+    if are_sys_data_matched:
+        if is_hvac_sys_cooling_type_dx(rmi_b, hvac_b_id):
+            if is_hvac_sys_heating_type_furnace(rmi_b, hvac_b_id):
+                is_baseline_system_3 = HVAC_SYS.SYS_3
+            elif is_hvac_sys_heating_type_fluid_loop(rmi_b, hvac_b_id):
+                is_baseline_system_3 = HVAC_SYS.SYS_3B
```

Comment:
It seems here is missing a `is_hvac_sys_fluid_loop_purchased_heating` to determine HVAC_SYS.SYS_3B

Reply:
Great catch! ```is_hvac_sys_fluid_loop_purchased_heating``` is added to the code.